src/main/java/uk/gov/dwp/dataworks/errors/CurrentKeyIdFailure.java

```diff
@@ -0,0 +1,11 @@
+package uk.gov.dwp.dataworks.errors;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.SERVICE_UNAVAILABLE)
+public class CurrentKeyIdFailure extends RuntimeException  {
```

Comment:
Not sure if this is due to a naming convention or similar, but would probably call this CurrentKeyIdException as it describes the _thing_ it is better, rather than the potential _implentation_ of the thing.

Reply:
Looking at this again and where it is used, the custom exceptions seem pretty specific (i.e. key missing) whereas the use cases are more generic (i.e. in KMSCurrentKeyProvider where it's any exception).

Would suggest either catching specific exceptions (i.e. in the example above, could catch InvalidKeyIdException) and throwing the custom ones or changing the custom ones to be more generic.

Catching specific exceptions feels like a nicer way of handling the errors.

Reply:
As discussed